api/v1alpha1/embedder.go

```diff
@@ -24,6 +24,32 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+func (e Embedder) Provider() Provider {
```

Comment:
maybe like Model will use IsLLMModel function, Provider should also provider Is3rdParty and IsProviderWorker function?

Reply:
provide type and switch case is good enough